pdns/sdig.cc

```diff
@@ -169,10 +169,10 @@ static void printReply(const string& reply, bool showflags, bool hidesoadetails,
          iter != edo.d_options.end(); ++iter) {
       if (iter->first == EDNSOptionCode::ECS) { // 'EDNS subnet'
         EDNSSubnetOpts reso;
-        if (getEDNSSubnetOptsFromString(iter->second, &reso)) {
-          cerr << "EDNS Subnet response: " << reso.source.toString()
-               << ", scope: " << reso.scope.toString()
-               << ", family = " << reso.scope.getNetwork().sin4.sin_family
+        if (EDNSSubnetOpts::getFromString(iter->second, &reso)) {
+          cerr << "EDNS Subnet response: " << reso.getSource().toString()
+               << ", scope: " << Netmask(reso.getSource().getNetwork(), reso.getScopePrefixLength()).toString()
```

Comment:
I see the same pattern in `dnsproxy.cc` so perhaps a `[[nodiscard]] const Netmask& EDNSSubnetOpts::getScope() const` method might make sense?

Reply:
Yes, wil do